prototypes/ItemRepeaterPrototypes/ItemsRepeaterExperiments/AttachedBehaviors/SelectionModelPage.xaml.cs

```diff
@@ -0,0 +1,102 @@
+﻿using ItemsRepeaterExperiments.Common;
+using Microsoft.UI.Xaml.Controls;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+
+// The Blank Page item template is documented at https://go.microsoft.com/fwlink/?LinkId=234238
+
+namespace ItemsRepeaterExperiments.AttachedBehaviors
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public sealed partial class SelectionModelPage : Page
+    {
+        private List<Team> teams = DataSourceCreator<Team>.CreateRandomizedList(100);
+        private SelectionModel selectionModel = new SelectionModel();
+        public SelectionModelPage()
+        {
+            this.InitializeComponent();
+            selectionModel.Source = teams;
+            selectionModel.SelectionChanged += SelectionModel_SelectionChanged;
+            selectionModel.SingleSelect = true;
+        }
+
+
+        /// <summary>
+        /// Attach properties to virtualized items and update containers
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        private void ItemsRepeater_ElementPrepared(Microsoft.UI.Xaml.Controls.ItemsRepeater sender, Microsoft.UI.Xaml.Controls.ItemsRepeaterElementPreparedEventArgs args)
+        {
+            new PointerBehaviors(args.Element as Control).Click += Item_Click;
```

Comment:
If you update this behavior to be attached similar to SelectionBehavior comment, it can be done in markup and we dont need the ElementPrepared handler anymore.

Reply:
I am not sure how we would listen to the events of those items. After all, we need to get a reference to the UI elements to attach to the listeners.

Reply:
When the attached property is set on an element like so
```<UserControl SelectionBehavior.IsEnabled='true' />``` 
The attached property callback will be called and you get the element it got attached to in the callback. You can hookup to the necessary event handlers there.

Reply:
Great idea, updated the PointerBehaviors.